src/main/java/com/chromediopside/service/ErrorService.java

```diff
@@ -34,12 +34,12 @@ String missingValues(LoginForm loginForm) {
 
     }
     Collections.sort(missingFields);
-    return String.join(", ", missingFields) + "!";
+    return String.join(", ", missingFields);
   }
 
   public ErrorResponse fieldErrors(LoginForm loginForm) {
     errorResponse.setStatus("error");
-    errorResponse.setMessage(MISSING_PARAMS_MESSAGE + missingValues(loginForm));
+    errorResponse.setMessage(MISSING_PARAMS_MESSAGE + missingValues(loginForm) + "!");
```

Comment:
if you'd like to change this to a more memory friendly solution, use other String concatenation way here. e.g. Java 8's `StringJoiner`: https://docs.oracle.com/javase/8/docs/api/java/util/StringJoiner.html